### PR TITLE
Retain denial reasons after advice has been countersigned

### PR DESCRIPTION
**Pull request by kevincarrogan**
### Aim

Stop denial reasons from being deleted.

[LTD-4030](https://uktrade.atlassian.net/browse/LTD-4030)

[LTD-4030]: https://uktrade.atlassian.net/browse/LTD-4030?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: api/cases/models.py [hunk 1]
```diff
@@ -486,21 +479,16 @@ def save(self, *args, **kwargs):
...
-                    )
+                denial_reasons = [denial_reason for denial_reason in old_advice.denial_reasons.all()]
                 old_advice.delete()
         except Advice.DoesNotExist:
             pass
 
         super(Advice, self).save(*args, **kwargs)
 
+        if denial_reasons:
+            self.denial_reasons.set(denial_reasons)
```
Comment (saruniitr) [inline api/cases/models.py:490]: If the new instance has `denial_reasons` then they are overwritten by the old one.

Comment (kevincarrogan) [reply api/cases/models.py:490]: Is there a real use case where this happens so I can write a test for this?

The problem here is that if I combine them there might be the case where someone is purposefully overwriting the denial reasons and combining them is wrong i.e. when someone is editing.

The biggest issue we've got at the moment is that we're just deleting them, which is what we need to resolve to stop this from happening.

Comment (saruniitr) [reply api/cases/models.py:490]: Users can edit their recommendation and change the criteria.
I think we need tests covering both cases.
- FCDO user gives refusal recommendation with some criteria and submits. He then decides to edit their recommendation and changes criteria so in this case we only want to save the new criteria

File: api/cases/tests/test_create_advice.py [hunk 2]
```diff
@@ -422,6 +427,35 @@ def test_countersign_advice_success(self):
         self.assertEqual(audit_qs.count(), 1)
         self.assertEqual(audit_qs.first().actor, self.gov_user)
 
+    def test_countersigning_retains_denial_reasons(self):
+        advice = Advice.objects.create(
+            user=self.gov_user,
+            case=self.case,
+            note="Advice",
```
Comment (saruniitr) [inline api/cases/tests/test_create_advice.py:434]: this should actually be set to `text` field.
`note` is where additional notes to exporter field

**Review by saruniitr — APPROVED** [review]

**Review by depsiatwal — APPROVED** [review]
lgtm

**Pull request merged**